packages/components/src/components/legends/as-legend-size-continuous-line/as-legend-size-continuous-line.tsx

```diff
@@ -0,0 +1,210 @@
+import { Component, Prop, State } from '@stencil/core';
+
+const TEXT_MARGIN = 4;
+const MIN_LINE_SIZE = 4;
+
+@Component({
+  shadow: false,
+  styleUrl: 'as-legend-size-continuous-line.scss',
+  tag: 'as-legend-size-continuous-line'
+})
+export class LegendSizeContinuousLine {
+  @Prop() public data: LegendData[];
+  @Prop() public orientation: 'horizontal' | 'vertical' = 'vertical';
+  @Prop() public size: number = 300;
+  @Prop() public leadingLineStrokeWidth: number = 0.5;
+  // Line height of the font, which is 12 with the original as-caption
+  @Prop() public textLineHeight: number = 12;
+
+  @Prop() public xMarginFactor: number = 0.1;
+  @Prop() public yMarginFactor: number = 0.1;
+
+  @Prop() public width: number = null;
+
+  @State() private rSize = 0;
+
+
+  public componentWillUpdate() {
+    this.parseSize();
+  }
+
+  public componentWillLoad() {
+    this.parseSize();
+  }
+
+  public render() {
+    const sortedData = this.data.slice().sort(
+      (first, second) => second.width - first.width
+    );
+
+    const max = sortedData[0].width;
```

Comment:
I would always use capital case for constant numbers: MAX, HALF, etc. 

Also, I'd use `X_OFFSET` and `Y_OFFSET` (instead of yOffset). If it's common to use `yOffset`, I'd use it then for both X and Y offsets.